quilter.py

```diff
@@ -0,0 +1,360 @@
+
+        for dir_name, subdir_list, file_list in os.walk(extract_folder):
+            for fname in file_list:
+                if str.endswith(fname, ".shp"):
+                    shp_list.append(os.path.join(dir_name, fname))
+
+        subprocess.run(shp_args)
+    
+    # raster merging
+    elif merge and raster:
+        print("\nCombining into one dataset...")
+
+        # Merge files by building VRT
+        vrt_list = []
+        vrt_path = os.path.join(tdir, final_name + str(os.getpid()) +
+                                ".vrt")
+        for dir_name, subdir_list, file_list in os.walk(extract_folder):
+            for fname in file_list:
+                if str.endswith(fname, raster_exts):
+                    vrt_list.append(os.path.join(dir_name, fname))
+
+        vrt_opts = gdal.BuildVRTOptions(resampleAlg='cubic')
+        vrt = gdal.BuildVRT(vrt_path, vrt_list, options=vrt_opts)
+        vrt = None  # releases file handle
+
+        if compress:
+            co=["bigtiff=yes", "compress=lzw", "tiled=yes"]
+        else:
+            co=["tiled=yes"]
+
+        # overwrite creation options for jpeg compression of aerial imagery
+        if len(vrt_list) > 0:
+            ds = gdal.Open(vrt_list[0], gdal.GA_ReadOnly)
+            md = ds.GetMetadata('IMAGE_STRUCTURE')
+            if 'COMPRESSION' in md and 'JPEG' in md["COMPRESSION"]:
+                co = ["bigtiff=yes", "compress=jpeg", "tiled=yes"]
+                gdal.SetConfigOption('COMPRESS_OVERVIEW', 'JPEG')
+                if 'YCbCr' in md['COMPRESSION']:
+                    co.append("photometric=ycbcr")
+                    gdal.SetConfigOption('PHOTOMETRIC_OVERVIEW', 'YCBCR')
+
+        # project if desired
+        if project:
+            print("\nProjecting {} to {}".format(raster_outpath, projection))
+            warp_opts = gdal.WarpOptions(dstSRS=projection, 
+                                         resampleAlg="cubic", 
+                                         format="GTiff", 
+                                         multithread=True,
+                                         creationOptions=co,
+                                         callback=gdal_progress_callback)
+            ds = gdal.Warp(raster_outpath, vrt_path, options=warp_opts)
+            ds = None  # releases file handle
+
+            print("\nBuilding overviews...")
+            dso = gdal.Open(raster_outpath, gdal.GA_ReadOnly)
+            dso.BuildOverviews("NEAREST", [2, 4, 8, 16], gdal_progress_callback)
+            dso = None
+        
+        # otherwise, just translate vrt to tif
+        else:
+            print("\nMerging into {}".format(raster_outpath))
+            trans_opts = gdal.TranslateOptions(format="GTiff", 
+                                               creationOptions=co, 
+                                               callback=gdal_progress_callback)
+            ds = gdal.Translate(raster_outpath, vrt_path, options=trans_opts)
+            ds=None
```

Comment:
what is your programming environment? We have some general linters and formatter settings we could share with you hopefully to avoid some of these types of formatting mistakes.

Reply:
Right now I'm running VS Code with their default Python environment. I've used Atom + Flake8 +PEP8 before, but I'm finding VS Code's IntelliSense to work better than Atom's implementation... though that could just be the way I had Atom set up. I'd love to see your guys' settings.

Reply:
Cool! Here's a vscode [.settings](https://github.com/steveoh/bulldozer/blob/master/.vscode/settings.json) file that I used recently. It is more or less descriptive of what we use. Here is the [yapf](https://github.com/steveoh/bulldozer/blob/master/.style.yapf) settings we use on most projects for formatting. And some [pylint](https://github.com/steveoh/bulldozer/blob/master/.pylintrc) settings for linting.